core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -49,6 +50,13 @@ public interface PackageConfig {
      */
     Optional<String> outputName();
 
+    /**
+     * The timestamp used as a reference for generating the packages (e.g. for the creation timestamp of ZIP entries).
+     * <p>
+     * The approach is similar to what is done by the maven-jar-plugin with `project.build.outputTimestamp`.
+     */
+    Optional<Instant> outputTimestamp();
```

Comment:
What happens if the user does not set the value? Is `project.build.outputTimestamp` propagated? I wonder if the JavaDoc should contain this kind of info.

Reply:
Ah, yes, I can add a comment. You actually don't need to set it in the Maven case, I propagate `project.build.outputTimestamp`.

As for Gradle, I'll see when I get a Maven build fully reproducible.